lighthouse-core/lib/tracehouse/main-thread-tasks.js

```diff
@@ -44,104 +44,268 @@ const {taskGroups, taskNameToGroup} = require('./task-groups.js');
 class MainThreadTasks {
   /**
    * @param {LH.TraceEvent} event
-   * @param {TaskNode} [parent]
+   * @param {Pick<LH.TraceEvent, 'ph'|'ts'>} [endEvent]
    * @return {TaskNode}
    */
-  static _createNewTaskNode(event, parent) {
+  static _createNewTaskNode(event, endEvent) {
+    const isCompleteEvent = event.ph === 'X' && !endEvent;
+    const isStartEndEventPair = event.ph === 'B' && endEvent && endEvent.ph === 'E';
+    if (!isCompleteEvent && !isStartEndEventPair) {
+      throw new Error('Invalid parameters for _createNewTaskNode');
+    }
+
+    const startTime = event.ts;
+    const endTime = endEvent ? endEvent.ts : event.ts + Number(event.dur || 0);
+
     const newTask = {
       event,
-      startTime: event.ts,
-      endTime: event.ph === 'X' ? event.ts + Number(event.dur || 0) : NaN,
-      parent: parent,
-      children: [],
+      startTime,
+      endTime,
+      duration: endTime - startTime,
 
       // These properties will be filled in later
+      parent: undefined,
+      children: [],
       attributableURLs: [],
       group: taskGroups.other,
-      duration: NaN,
       selfTime: NaN,
     };
 
-    if (parent) {
-      parent.children.push(newTask);
-    }
-
     return newTask;
   }
 
   /**
-   * @param {LH.TraceEvent[]} mainThreadEvents
+   *
+   * @param {TaskNode} currentTask
+   * @param {{startTime: number}} nextTask
    * @param {PriorTaskData} priorTaskData
+   * @param {Array<LH.TraceEvent>} reverseEventsQueue
+   */
+  static _assignAllTimerInstallsBetweenTasks(
+      currentTask,
```

Comment:
I respectfully disagree, you just must not have seen very much nicely formatted JS ;)

All standalone functions with multiple lines in our codebase are formatted this way except 1, and that's only because it was [authored by an external contributor two years ago](https://github.com/GoogleChrome/lighthouse/commit/890103468bc724cc221e47bbbb91a21fb35efbe4#diff-e2079966cf5431d4d65b3c6dbd91b25cR37). This is also the formatting approach taken by [prettier](https://prettier.io/) which has become somewhat of a defacto standard style across new JS projects in the community.

I rest my case 👩‍⚖ 😄

Reply:
yeah i prefer this too, but I disagree with any argument using "prettier does it" since I have grown to dislike how "prettier" code looks =P

Reply:
> I disagree with any argument using "prettier does it" since I have grown to dislike how "prettier" code looks =P

That's fine but no matter how you feel about prettier it is clear evidence this style is present in the javascript community and should not feel like only cpp.

Reply:
motion to suppress evidence successful 🎉